examples/Cassie/osc_jump/jumping_event_based_fsm.cc

```diff
@@ -0,0 +1,187 @@
+#include "examples/Cassie/osc_jump/jumping_event_based_fsm.h"
+
+#include <drake/lcmt_contact_results_for_viz.hpp>
+#include "dairlib/lcmt_contact_mujoco.hpp"
+
+using dairlib::systems::OutputVector;
+using drake::multibody::MultibodyPlant;
+using drake::systems::BasicVector;
+using drake::systems::Context;
+using drake::systems::DiscreteUpdateEvent;
+using drake::systems::DiscreteValues;
+using drake::systems::EventStatus;
+using Eigen::VectorXd;
+using std::string;
+using std::vector;
+
+namespace dairlib {
+namespace examples {
+
+JumpingEventFsm::JumpingEventFsm(const MultibodyPlant<double>& plant,
+                                 const vector<double>& transition_times,
+                                 bool contact_based, double delay_time,
+                                 FSM_STATE init_state, SIMULATOR simulator_type)
+    : plant_(plant),
+      transition_times_(transition_times),
+      contact_based_(contact_based),
+      transition_delay_(delay_time),
+      init_state_(init_state),
+      simulator_type_(simulator_type) {
+  state_port_ =
+      this->DeclareVectorInputPort(OutputVector<double>(plant.num_positions(),
+                                                        plant.num_velocities(),
+                                                        plant.num_actuators()))
+          .get_index();
+
+  // Configure the contact info port for the particular simulator
+  if (simulator_type_ == DRAKE) {
```

Comment:
It would be nice if both simulators looked the same to this class, but maybe that's something to be fixed in the future.

Reply:
I agree. Right now I output the lumped contact force/moment from mujoco. I'll look into either making mujoco output the `drake::lcmt_contact_results_for_viz` or pre-process the lcm message from mujoco to make it cleaner.